### PR TITLE
chore: deprecate address.streetPrefix and suffix

### PR DESCRIPTION
**Pull request by ST-DDT**
The street prefix and suffix methods only generate parts of the data and not all of the locales support these.
IMO the methods aren't useful by themselves and thus should be removed.

### CODE REVIEW

**Comment by xDivisionByZerox** [comment]
IMO either keep them as util functions (for custom tooling) or remove them completely, like don't advertise the usage with `faker.fake()`. Doesn't make sense to me to tell the user "hey can still use this, but not as nice as it was for now" (write a "complex" non-typesafe string).

**Comment by ST-DDT** [comment]
> IMO either keep them as util functions (for custom tooling) or remove them completely, like don't advertise the usage with `faker.fake()`.

I do want to remove them, but I want to give everyone a chance to migrate.
I could switch to `faker.helpers.arrayElement(faker.definitions.address.street_prefix)`, but that is way longer.
The current usages of these methods are mostly limited to `faker.fake()`, so thats why I have chosen that one.

**Comment by xDivisionByZerox** [comment]
> I do want to remove them, but I want to give everyone a chance to migrate.
> I could switch to `faker.helpers.arrayElement(faker.definitions.address.street_prefix)`, but that is way longer.
> The current usages of these methods are mostly limited to `faker.fake()`, so thats why I have chosen that one.

Nah, I think you misunderstood me. If we deprecate these functions, why leave the definitions?

File: src/modules/address/index.ts [hunk 1]
```diff
@@ -216,10 +216,21 @@ export class Address {
...
    * @example
    * faker.address.streetSuffix() // 'Streets'
+   *
+   * @deprecated
    */
   streetSuffix(): string {
+    deprecated({
+      deprecated: 'faker.address.streetSuffix()',
+      proposed:
+        "faker.address.street() or faker.fake('{{address.street_suffix}}')",
```
Comment (xDivisionByZerox) [inline src/modules/address/index.ts:230]:
```suggestion
        "faker.address.street()",
```

I wouldn't want to advertise the direct usage of definitions.

**Comment by ST-DDT** [comment]
> If we deprecate these functions, why leave the definitions?

To give users a hint how they should structure their definitions to be compatible with our format/our existing locales.
If possible I still want all locales to have the same set of possible files.
But there might be other/better ways to achieve that (e.g. a file-name whitelist, so it's easier to tell when a file is added that doesn't match our existing files).

File: test/random.spec.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ const seededRuns = [
       locale: 'ur',
       numeric: '9',
       word: 'Division',
-      words: 'Turnpike Frozen Handcrafted',
+      words: 'functionalities Frozen Baby',
```
Comment (Shinigami92) [inline test/random.spec.ts:36]: 🥶

**Review by Shinigami92 — APPROVED** [review]
IMO it is okay to remove / hide this from the end-user as I would not know what a street suffix/prefix would even be 🤷
We also didn't explain it anywhere but just provide it
In the end Faker should be a tool to generate meaningful data, and streetnames are totally okay, but what is a street prefix/suffix?

**Review by xDivisionByZerox — DISMISSED** [review]

**Review by import-brain — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by pkuczynski — DISMISSED** [review]

File: src/modules/address/index.ts [hunk 3]
```diff
@@ -218,10 +218,20 @@ export class Address {
   /**
    * Returns a random localized street suffix.
    *
+   * @see faker.address.street
+   *
    * @example
    * faker.address.streetSuffix() // 'Streets'
+   *
+   * @deprecated
```
Comment (Shinigami92) [inline src/modules/address/index.ts:226]: I might be wrong, but did we discuss to start writing human readable messages on deprecation JSDoc tags?
https://github.com/faker-js/faker/issues/976

Comment (xDivisionByZerox) [reply src/modules/address/index.ts:226]: Yes, good catch. See our [contribution guidelines, section deprecation workflow]:

> Afterwards, add a @deprecated parameter to the end of the JSDoc with a human readable description message with a suitable replacement for the deprecated function.

Comment (ST-DDT) [reply src/modules/address/index.ts]: Done

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**